hummingbot/market/kraken/kraken_api_order_book_data_source.py

```diff
@@ -0,0 +1,278 @@
+    PING_TIMEOUT = 10.0
+
+    _haobds_logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._haobds_logger is None:
+            cls._haobds_logger = logging.getLogger(__name__)
+        return cls._haobds_logger
+
+    def __init__(self, symbols: Optional[List[str]] = None):
+        super().__init__()
+        self._symbols: Optional[List[str]] = symbols
+
+    @classmethod
+    @async_ttl_cache(ttl=60 * 30, maxsize=1)
+    async def get_active_exchange_markets(self) -> pd.DataFrame:
+        async with aiohttp.ClientSession() as client:
+            exchange_response = await client.get(KRAKEN_MARKETS_URL)
+            exchange_response: aiohttp.ClientResponse = exchange_response
+            if exchange_response.status != 200:
+                raise IOError(f"Error fetching Kraken exchange information. "
+                              f"HTTP status is {exchange_response.status}.")
+            exchange_data = await exchange_response.json()
+            trading_pairs = {
+                trading_pair: {
+                    "symbol": f"{info['base']}/{info['quote']}",
+                    "baseAsset": info['base'],
+                    "quoteAsset": info['quote'],
+                    "altname": info['altname']
+                } for trading_pair, info in exchange_data["result"].items()
+            }
+            pairs = [info["altname"] for info in exchange_data["result"].values()]
+
+            params = '&pair='
+            params += ','.join(pairs)
+            exchange_data_ = []
+            exchange_data_.append(exchange_data['result'])
+            market_response = await client.request(url=KRAKEN_TICKER_URL, params=params, method='GET')
+            market_data = await market_response.json()
+            if market_response.status != 200:
+                raise IOError(f"Error fetching Kraken markets information. "
+                              f"HTTP status is {market_response.status}.")
+            market_data: List[Dict[str, Any]] = [
+                {**item[1], **trading_pairs[item[0]]}
+                for item in market_data['result'].items()
+                if item[0] in trading_pairs
+            ]
+            for colum in market_data:
+                colum['quoteVolume'] = colum['v'][1]
+                colum.pop('v')
+                colum.pop('a')
+                colum.pop('b')
+                colum['close'] = colum['c'][0]
+                colum.pop('c')
+                colum['hcolumgh'] = colum['h'][0]
+                colum.pop('h')
+                colum['low'] = colum['l'][0]
+                colum.pop('l')
+                colum['open'] = colum['o'][0]
+                colum.pop('o')
+            all_markets: pd.DataFrame = pd.DataFrame.from_records(data=market_data, index="symbol")
+            btc_price: float = float(all_markets.loc["XBTUSD"]["close"])
+            eth_price: float = float(all_markets.loc["ETHUSD"]["close"])
```

Comment:
As mentioned earlier, this needs to change to adapt to Kraken's naming convention for `BTC`, `USD` and `ETH`